src/include/ldcs_api_pipe.h

```diff
@@ -35,8 +35,8 @@ char *ldcs_get_connection_string_pipe(int fd);
 int ldcs_register_connection_pipe(char *connection_str);
 
 /* internal */
-int _ldcs_write_pipe(int fd, const void *data, int bytes );
-int _ldcs_read_pipe(int fd, void *data, int bytes, ldcs_read_block_t block );
+size_t _ldcs_write_pipe(int fd, const void *data, size_t bytes, int *err );
+size_t _ldcs_read_pipe(int fd, void *data, size_t bytes, ldcs_read_block_t block, int *err );
```

Comment:
Similar concerns here, though there's larger issues with adding err to ldcs_read_pipe and ldcs_write_pipe. The implementations below set err even when there's soft errors that the functions recover from. But the calling code treats all contents of *err as a hard error. 
Again, suggest making the return ssize_t and using a -1 return on error.